src/test/java/com/google/googleidentity/oauth2/authorizationcode/AuthorizationCodeRequestHandlerTest.java

```diff
@@ -0,0 +1,350 @@
+
+/**
+ * Tests for {@link AuthorizationCodeRequestHandler}
+ */
+public class AuthorizationCodeRequestHandlerTest {
+
+    private UserSession userSession;
+
+    private static final String CLIENTID = "client";
+    private static final String CLIENTID1 = "client1";
+    private static final String SECRET = "secret";
+    private static final String REDIRECT_URI = "http://www.google.com";
+
+    private static final String REDIRECT_URI1 = "http://www.facebook.com";
+
+    private static final String LINE = System.lineSeparator();
+
+
+    private static final ClientDetails CLIENT =
+            ClientDetails.newBuilder()
+                    .setClientId(CLIENTID)
+                    .setSecret(Hashing.sha256()
+                            .hashString(SECRET, Charsets.UTF_8).toString())
+                    .addScopes("read")
+                    .setIsScoped(true)
+                    .addRedirectUris(REDIRECT_URI)
+                    .addGrantTypes(OAuth2Constants.GrantType.AUTHORIZATION_CODE)
+                    .build();
+
+    private static final String USERNAME = "username";
+    private static final String PASSWORD = "password";
+    private static final String STATE = "state";
+
+    private static final UserDetails USER =
+            UserDetails.newBuilder()
+                    .setUsername(USERNAME)
+                    .setPassword(Hashing.sha256()
+                            .hashString(PASSWORD, Charsets.UTF_8).toString())
+                    .build();
+
+    AuthorizationCodeRequestHandler authorizationCodeRequestHandler;
+
+    AuthorizationCodeService authorizationCodeService;
+
+    OAuth2TokenService oauth2TokenService;
+
+
+    OAuth2Request TEST_REQUEST =
+            OAuth2Request.newBuilder()
+                    .setRequestAuth(
+                            OAuth2Request.RequestAuth.newBuilder()
+                                    .setClientId(CLIENTID)
+                                    .setUsername(USERNAME)
+                                    .build())
+                    .setRequestBody(
+                            OAuth2Request.RequestBody.newBuilder()
+                                    .setIsScoped(true)
+                                    .addAllScopes(CLIENT.getScopesList())
+                                    .setResponseType(OAuth2Constants.ResponseType.CODE)
+                                    .setRefreshable(true)
+                                    .setGrantType(OAuth2Constants.GrantType.AUTHORIZATION_CODE)
+                                    .build())
+                    .setAuthorizationResponse(
+                            OAuth2Request.AuthorizationResponse.newBuilder()
+                                    .setState(STATE)
+                                    .setRedirectUri(REDIRECT_URI)
+                                    .build())
+                    .build();
+
+    OAuth2Request TEST_REQUEST1 =
+            OAuth2Request.newBuilder()
```

Comment:
The indent looks not standard, you may later use a dedicated pull request to fix the indent by using intellij-java-google-style.xml 
https://github.com/google/styleguide/blob/gh-pages/intellij-java-google-style.xml
https://gerrit.wikimedia.org/r/Documentation/dev-intellij.html#_code_style_settings